File: ts/problem_solving/kata/clean_string/solution_test.ts
import {assertEquals} from "../../../dev_deps.ts"
import {solution} from "./solution.ts"

Deno.test("solution it works", () => {
  assertEquals(solution("abc#d##c"), "ac")
  assertEquals(solution("abc##d######"), "")
  assertEquals(solution("#######"), "")
})

File: __algos/go/problem_solving/kata/clean_string/solution_test.go
package clean_string

import (
	"fmt"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestCleanString(t *testing.T) {

	type test struct {
		input    string
		expected string
	}

	tests := []test{
		{input: "abc#d##c", expected: "ac"},
		{input: "abc##d######", expected: ""},
		{input: "#6+yqw8hfklsd-=-f", expected: "6+yqw8hfklsd-=-f"},
	}

	for _, tc := range tests {
		got := solution(tc.input)
		assert.Equal(t, tc.expected, got, fmt.Sprintf("input: %s", tc.input))
	}

}
